src/main/java/de/cyklon/reflection/entities/impl/ReflectClassImpl.java

```diff
@@ -104,30 +105,28 @@ public <E extends Enum<E>> List<E> getEnumConstants() throws IllegalStateExcepti
 
 	@Override
 	public boolean isPrimitive() {
-		return clazz != null && clazz.isPrimitive();
+		return clazz.isPrimitive();
 	}
 
 	@Override
 	public boolean isArray() {
-		return clazz == null || clazz.isArray();
+		return clazz.isArray();
 	}
 
 	@Override
 	public boolean isEnum() {
-		return clazz != null && clazz.isEnum();
+		return clazz.isEnum();
 	}
 
 	@NotNull
 	@Override
 	public EnumSet<Modifier> getModifiers() {
-		if (clazz == null) return EnumSet.noneOf(Modifier.class);
 		return Modifier.parse(clazz.getModifiers());
 	}
 
 	@Nullable
 	@Override
 	public ReflectClass<?> getParentClass() {
-		if (clazz == null) return null;
 		return clazz.getSuperclass() == null ? null : wrap(clazz.getSuperclass());
```

Comment:
Create a variable for clazz.getSuperclass() so that you don't have to call this method twice. This can improve performance a little.